internal/fs/inode/file.go

```diff
@@ -178,38 +180,74 @@ func (f *FileInode) clobbered(ctx context.Context) (b bool, err error) {
 	return
 }
 
-// Ensure that f.content != nil
+// Helper method to open a reader
+// (should only be called by methods that have locks)
 //
 // LOCKS_REQUIRED(f.mu)
-func (f *FileInode) ensureContent(ctx context.Context) (err error) {
-	// Is there anything to do?
-	if f.content != nil {
-		return
-	}
-
+func (f *FileInode) openReader(ctx context.Context) (io.ReadCloser, error) {
 	// Open a reader for the generation we care about.
 	rc, err := f.bucket.NewReader(
 		ctx,
 		&gcs.ReadObjectRequest{
 			Name:       f.src.Name,
 			Generation: f.src.Generation,
 		})
-
 	if err != nil {
 		err = fmt.Errorf("NewReader: %w", err)
-		return
 	}
+	return rc, err
+}
 
-	// Create a temporary file with its contents. The temp file
-	// ensures to call Close() on the rc.
-	tf, err := f.contentCache.NewTempFile(rc)
-	if err != nil {
-		err = fmt.Errorf("NewTempFile: %w", err)
-		return
-	}
+// Ensure that content exists and is not stale
+//
+// LOCKS_REQUIRED(f.mu)
+func (f *FileInode) ensureContent(ctx context.Context) (err error) {
+	if f.localFileCache {
+		// Fetch content from the cache
+		// we validate generation numbers here again
```

Comment:
Be consistent with the comments. All the comments start with a capitalized letter.